manual/sphinx/user_docs/variable_init.rst

```diff
@@ -328,22 +328,20 @@ to ``generate`` in the ``Context`` object.
 
   Field3D shear = ...; // Value calculated in BOUT++
   
-  FieldFactory factory(mesh);
-  auto gen = factory->parse("model:viscosity");
+  Options& model_options = options["model"];
+  auto str = model_options["viscosity"].doc("Viscosity").as<std::string()>;
+
+  auto gen = FieldFactory::get()->parse(str, &model_options);
 
   Field3D viscosity;
   viscosity.allocate();
-  
-  BOUT_FOR(i, viscosity.region("RGN_ALL")) {
-    viscosity[i] = gen->generate(bout::generator::Context(i, CELL_CENTRE, mesh, 0.0)
-                                   .set("shear", shear[i]));
+
+  BOUT_FOR(i, viscosity.getRegion("RGN_ALL")) {
+    viscosity[i] = gen->generate(bout::generator::Context().set("shear", shear[i]));
```

Comment:
This will use the default `Context()` constructor (https://github.com/boutproject/BOUT-dev/blob/master/include/bout/sys/generator_context.hxx#L19) so "x", "y", "z", "t" will not be defined. Recommend instead using the original `Context(i, CELL_CENTRE, mesh, 0.0)`